src/vivilux/layers.py

```diff
@@ -174,6 +176,24 @@ def RunProcesses(self):
     def AddProcess(self, process: Process):
         process.AttachLayer(self)
 
+    def AddMonitor(self, monitor: Monitor):
+        self.monitors.append(monitor)
+
+    def UpdateMonitors(self):
+        self.UpdateSnapshot()
```

Comment:
It appears `UpdateSnapshot` is called here and also in `StepTime`, is this intentional?

Reply:
No, that wasn't intentional. I forgot to delete `UpdateSnapshot` from `UpdateMonitors`.